display_providers.py

```diff
@@ -25,6 +27,19 @@ def get_seconds(self) -> int:
         pass
 
 
+class LocalTimeProvider(TimeProvider):
+
+    def get_hours(self) -> int:
+        hour = datetime.datetime.now().hour
+        return hour if hour < 13 else hour - 12
+
+    def get_minutes(self) -> int:
+        return datetime.datetime.now().minute
+
+    def get_seconds(self) -> int:
+        return datetime.datetime.now().second
+
+
```

Comment:
This was moved from main.py
Partly this is because main.py is a bit cluttered and big, but mostly it's because I can't test anything in main.py because when I import anything from main then the whole system starts up.
I guess the main script should be refactored to allow testing, but for now I took the easier route.